var/spack/repos/builtin/packages/py-pyqt5/package.py

```diff
@@ -46,5 +66,35 @@ def configure_args(self):
                          '--qsci-api-destdir', self.prefix.share.qsci])
         return args
 
+    @when('@5.14:')
+    def build(self, spec, prefix):
+        with working_dir(self.stage.source_path):
+            sip = Executable(join_path(spec['py-sip'].prefix.bin, 'sip-install'))
+            sip(*self.build_args())
```

Comment:
Instead of using `sip-install`, is it possible to build with `pip`? Since sip is now a build-backend listed in `pyproject.toml` it should theoretically work with other Python build tools like pip and build. 

Then, once the older versions are removed, we can directly subclass `PythonPackage` and delete `SIPPackage` entirely. I think we would have to deprecate `py-pyqt4` first too. There's also a new `py-pyqt6` package on PyPI.

Conversely, is it possible to build older versions with `sip-install`? What about `py-pyqt4`? Just trying to see if we can keep things uniform and/or reduce the number of build systems we support.